src/formatDate.js

```diff
@@ -8,7 +8,53 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let fromSeparator = '';
+
+  for (const arg of fromFormat) {
+    if (arg !== 'YYYY' || arg !== 'YY' || arg !== 'MM' || arg !== 'DD') {
+      fromSeparator = arg;
+    }
+  }
+
+  const dateParts = date.split(fromSeparator);
+
+  let year = '';
+  let month = '';
+  let day = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = dateParts[i];
+    } else if (fromFormat[i] === 'MM') {
+      month = dateParts[i];
+    } else {
+      day = dateParts[i];
+    }
+  }
+
+  if (toFormat.includes('YYYY') && fromFormat.includes('YY')) {
+    year = year < 30 ? `20${year}` : `19${year}`;
+  } else if (toFormat.includes('YY') && fromFormat.includes('YYYY')) {
+    year = year.slice(-2);
+  }
+
+  const newDateParts = [];
+
+  let toSeparator;
+
+  for (let i = 0; i < toFormat.length; i++) {
+    if (toFormat[i] === 'YYYY' || toFormat[i] === 'YY') {
+      newDateParts.push(year);
+    } else if (toFormat[i] === 'MM') {
+      newDateParts.push(month);
+    } else if (toFormat[i] === 'DD') {
+      newDateParts.push(day);
+    } else {
+      toSeparator = toFormat[i];
```

Comment:
The logic for determining the `toSeparator` assumes that the separator is not part of the format strings, which might not always be the case. Ensure that separators are correctly identified and handled, even if they are not explicitly included as the last element in the format arrays.